src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -458,31 +466,53 @@ declare module 'react' {
 
 const uploadItems = async (props: DriveExplorerProps, rootList: IRoot[], files: File[]) => {
   const { dispatch, currentFolderId, onDragAndDropEnd } = props;
-  if (files.length) {
-    // files where dragged directly
-    await dispatch(
-      storageThunks.uploadItemsThunkNoCheck({
-        files,
-        parentFolderId: currentFolderId,
-        options: {
-          onSuccess: onDragAndDropEnd,
-        },
-      }),
-    );
-  }
 
-  if (rootList.length) {
-    for (const root of rootList) {
+  let totalFilesToUpload: number;
+
+  totalFilesToUpload = files.length;
+
+  const countTotalItemsToUpload = (rootList) => {
```

Comment:
rootList is not typed, it is of type `IRoot[]`

Reply:
Mutating a variable outside the function -in this case `totalFilesToUpload`- is not the best practice.

Having a function that is self contained, can be a better practice. If we ever want to refactor and re-use this function in other context for IRoot types, we would be set to export it and use it right where needed.

Refactor this code block so that we only have:

`const countTotalItemsToUpload = files.length + countTotalItemsInIRoot(rootList)`

Where `countTotalItemsInIRoot` is a recursive function that returns the total number of items in an IRoot.